File: OsmAnd-java/src/main/java/net/osmand/reviews/Review.java
package net.osmand.reviews;

import java.net.URI;
import java.time.LocalDate;

/**
 * @param id      the review id, source system-specific
 * @param opinion a written opinion
 * @param rating  a numerical (star) rating in the range 1-100
 * @param author  the review author
 * @param date    the review date
 * @param link    a link to the review in the source system
 */
public record Review(
        String id,
        String opinion,
        int rating,
        String author,
        LocalDate date,
        URI link
) {
}

File: OsmAnd/src/net/osmand/plus/reviews/Review.kt
package net.osmand.plus.reviews

import java.time.LocalDate

/**
 * A POI review.
 * @property rating the review rating, a number between 1 and 100, inclusive
 * @property opinion optional opinion text, might be in any language
 * @property date the date when the review was last edited
 * @property author the review author's nickname
 */
data class Review(
    val rating: Int,
    val opinion: String?,
    val author: String,
    val date: LocalDate,
)
